docs/reference/components/transforms/remap.cue

```diff
@@ -0,0 +1,124 @@
+package metadata
+
+components: transforms: remap: {
+  title: "Remap"
+  description: "Maps and parses log fields using a light-weight, fast, and robust syntax."
+
+  _features: {
+  }
+
+  classes: {
+    commonly_used: true
+    function: "map"
+  }
+
+  statuses: {
+    development: "beta"
+  }
+
+  support: {
+    platforms: {
+      "aarch64-unknown-linux-gnu": true
+      "aarch64-unknown-linux-musl": true
+      "x86_64-pc-windows-msv": true
+      "x86_64-unknown-linux-gnu": true
+      "x86_64-unknown-linux-musl": true
+    }
+
+    requirements: []
+    warnings: []
+  }
+
+  configuration: {
+    source: {
+      common: true
+      description: "The remap source/instruction set to execute for each event"
+      required: true
+      type: string: {
+        examples: [
+          """
+          .type = "foo",
+          .new_field = .old_field * 2
+          del(.old_field)
+          """
+        ]
+      }
+    }
+  }
+
+  examples: log: [
+    {
+      title: "Add, Rename, & Remove Fields"
+      configuration: {
+        source: #"""
+                .new_field = "new value"
+                .new_name = .old_name
+                del(.old_name)
+                """#
+      }
+      input: {
+        old_name: "old value"
+      }
+      output: {
+        new_field: "new value"
+        new_name: "old value"
+      }
+    },
+    {
+      title: "Parse JSON"
+      configuration: {
+        source: #"""
+                message = del(.message)
+                . = parse_json(message)
```

Comment:
We currently don't have support for variables in the language.

We could obviously add it, if we wanted to, as it would remove the need to use the existing event as some kind of variable store (with all the problems that introduces, such as forgetting to delete a temporary value from the event, or overwriting an existing field with a temporary value).

If we were to add it, I would definitely advise not using `variable = value`, but have it explicitly be defined as `var variable = value`, as that would prevent accidental assignment to variables when you meant to add it to a path, e.g.:

```javascript
.foo = "bar"
foo = "bar" // confusing to the end-user if the field "foo" is missing from the event
```

```javascript
.foo = "bar"
foo = "bar" // parse error
var foo = "bar" // explicitly defined as a variable
```

This is probably worth a discussion separate from this PR though, but wanted to make sure you were aware the syntax you currently wrote is unsupported.